app/jobs/populate_register_data_in_db_job.rb

```diff
@@ -0,0 +1,117 @@
+class PopulateRegisterDataInDbJob < ApplicationJob
+  queue_as :default
+
+  def initialize
+    @registers_client ||= RegistersClient::RegisterClientManager.new(cache_duration: 60)
+  end
+
+  def bulk_remove_existing_records(register, entry_type, record_keys)
+    unless record_keys.empty?
+      Record.where(spina_register_id: register.id, entry_type: entry_type, key: record_keys).delete_all
```

Comment:
Should we use `destroy_all ` instead? https://apidock.com/rails/ActiveRecord/Relation/destroy_all

Reply:
`destroy_all` calls `destroy` on associations right (i.e. anything that's linked to the thing you're deleting by a foreign key). There are no relationships in this DB as far as I'm aware (entries and records store the item inside of themselves).